src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/ListenerPrimary.cs

```diff
@@ -179,5 +184,53 @@ private struct FILE_COMPLETION_INFORMATION
                 }, this).ConfigureAwait(false);
             }
         }
+
+        private class PipeReadContext
+        {
+            private readonly ListenerPrimary _listener;
+            private readonly IntPtr _bufPtr;
+            private int _bytesRead;
+
+            public PipeReadContext(ListenerPrimary listener)
+            {
+                _listener = listener;
+                _bufPtr = Marshal.AllocHGlobal(8);
```

Comment:
Why? If we never unpin it, isn't that even worse? It would also be inconsistent within the same class. I don't think we should change `_fileCompletionInfoPtr` right now. And I think if we do change to pinning, we should change both together.